op-node/rollup/derive/span_batch_tx.go

```diff
@@ -168,6 +187,23 @@ func (tx *spanBatchTx) convertToFullTx(nonce, gas uint64, to *common.Address, ch
 			R:          R,
 			S:          S,
 		}
+	case types.SetCodeTxType:
+		setCodeTxInner := tx.inner.(*spanBatchSetCodeTxData)
+		inner = &types.SetCodeTx{
+			ChainID:    uint256.MustFromBig(chainID),
+			Nonce:      nonce,
+			GasTipCap:  setCodeTxInner.GasTipCap,
+			GasFeeCap:  setCodeTxInner.GasFeeCap,
+			Gas:        gas,
+			To:         *to,
```

Comment:
We are sure this can never be `nil`?

Reply:
This is related to the other PR this was stacked on https://github.com/ethereum-optimism/optimism/pull/14197

Reply:
This is a very good observation @tynes I believe the current format could be hacked to make this nil. The contract creation bit could be set to 1 for SetCode transactions. We need to decide how to handle this case. We probably can just return an error.